<img src="https://user-images.strikinglycdn.com/res/hrscywv4p/image/upload/c_limit,fl_lossy,h_300,w_300,f_auto,q_auto/1266110/Logo_wzxi0f.png" style="float: left; margin: 20px; height: 55px">



## Useful Operators

#### 1. Print first 5 numbers using range function

In [4]:
r = list(range(1,6))
print(r)

[1, 2, 3, 4, 5]


#### 2. Print integers within given start and stop number using range() function

In [7]:
list(range(5))

[0, 1, 2, 3, 4]

#### 3. Create a list of integers between 100 and 126 included

In [8]:
list(range(100,127))

[100,
 101,
 102,
 103,
 104,
 105,
 106,
 107,
 108,
 109,
 110,
 111,
 112,
 113,
 114,
 115,
 116,
 117,
 118,
 119,
 120,
 121,
 122,
 123,
 124,
 125,
 126]

#### 4.Print all odd numbers between 23 and 64 using range()

In [9]:
list(range(23,65,2))

[23,
 25,
 27,
 29,
 31,
 33,
 35,
 37,
 39,
 41,
 43,
 45,
 47,
 49,
 51,
 53,
 55,
 57,
 59,
 61,
 63]

#### 5. Create a list with all the even numbers between 80 and 124 included.

In [10]:
list(range(80,125,2))

[80,
 82,
 84,
 86,
 88,
 90,
 92,
 94,
 96,
 98,
 100,
 102,
 104,
 106,
 108,
 110,
 112,
 114,
 116,
 118,
 120,
 122,
 124]

#### 6. Use enumerate to return the elemnts from the list where their index is greater than 3. Add them to a new list.

In [12]:
names = ['Pol','Tabixe','Pablo','Paco','Adri','Hector','Manu','Carlos','Ibrahim','Javier','Jesus']

In [13]:
new_names = []

for x, y in enumerate(names):
    if x > 3:
        new_names.append(y)

print(new_names)
    

['Adri', 'Hector', 'Manu', 'Carlos', 'Ibrahim', 'Javier', 'Jesus']


#### 7. Use enumerate to substract 2 to the numbers in the list that have and even index. Return the same list updated.

In [14]:
numbers = [2,44,564,0.3,45,6/7,9,15]

In [19]:
for x, y in enumerate(numbers):
    if x % 2 == 0:
        numbers[x]= y - 2
        

print(numbers)

[0, 44, 562, 0.3, 43, 0.8571428571428571, 7, 15]


#### 8. Use zip to group each variable on the following list based on their index

In [21]:
list1 = [15,2.5,3/6,334,5]
list2 = ['t','n','o','a','e']

In [25]:
list(zip(list1,list2))


[(15, 't'), (2.5, 'n'), (0.5, 'o'), (334, 'a'), (5, 'e')]

#### 9. Write a program which returns does elements that appear in both lists. Add them to a new list and sort it descending.

In [27]:
alpha = [6,7,33,6.5,908.9,34,2*2]
beta = [4,32,66,7,12+22]

In [31]:
repeated = []

for x in alpha:
    if x in beta:
        repeated.append(x)
        
print(repeated)

[7, 34, 4]


#### 10. Write a program which returns a list with does elements in the first list that match the length of the elements in the second list.

In [32]:
bv = ['Pol','Tabixe','Pablo','Paco','Adri','Hector','Manu','Carlos','Ibrahim','Javier','Jesus']
bv_num = [2,3,67,5,1,0]

In [35]:
elem = []

bv[int(len(bv_num))-1]

    

'Hector'

#### 11. Print a random number from a range between 100 and 4000.

In [39]:
from random import randint
randint(100,4001)

2291

#### 12. Print a shuffled version of the list provided.

In [41]:
numbers2 = [2,44,564,0.3,45,6/7,9,15]

In [44]:
from random import shuffle
shuffle(numbers2)
print(numbers2)

[9, 0.3, 0.8571428571428571, 15, 44, 2, 564, 45]


#### 13. Print an even random number between 0 and 10.

In [71]:
var = range(0,11,2)
for x in var:
    y = randint(0,10)
    if y%2==0:
        print(y)
        break

8


#### 14. Print a random element from a list

In [72]:
shuffle(numbers2)
print(numbers2[0])

0.8571428571428571


#### 15. Print a sample of 4 numbers from the list provided

In [73]:
sample = [2,5,8,9,12,66,32,56,554]

In [74]:
shuffle(sample)
print(sample[:4])

[9, 8, 32, 56]


#### 16. Create a program that asks for you name

In [75]:
input('please enter your name')

please enter your namejjj


'jjj'

#### 17. Create a program that asks for you name, age, location and returns the following output: Hi my name is _name_, I am _age_ years old and I live in _location_. Use any of the formatting you saw on day 1.

In [78]:
name = input('name?')
age = input('age?')
location = input('location?')

print(f"Hi my name is {name}, I am {age} years old and I live in {location}")

name?r
age?r
location?r
Hi my name is r, I am r years old and I live in r


#### 18. Create a program that asks for you name, age and returns the year were you will be 100 years old. 

#### 19. Write a single function called suma that returns the sum of the following examples withoug using more than one variable. Use *args

In [79]:
(3,5)
(4,5,6,7)
(1,2,3,5,6)

(1, 2, 3, 5, 6)

In [80]:
def suma(*args):
    return sum(args)
suma(3,5)
suma(4,5,6,7)
suma(1,2,3,5,6)

17

#### 20. Write a single function called multi_suma that returns the following. Use *args.

 - multi_suma(2,3,5) = ((2x3)+(2x5)) = 16
 - multi_suma(2,3,5,5) = ((2x3)+(2x5)+(2x5)) = 26
 - multi_suma(2,55,2,9) = ((2x55)+(2x5)+(2x9)) = 132